extensions/panache/mongodb-panache/deployment/src/test/java/io/quarkus/mongodb/panache/DuplicateIdMongoEntityTest.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.mongodb.panache;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.platform.launcher.Launcher;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder;
+import org.junit.platform.launcher.core.LauncherFactory;
+import org.junit.platform.launcher.listeners.SummaryGeneratingListener;
+import org.junit.platform.launcher.listeners.TestExecutionSummary;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class DuplicateIdMongoEntityTest {
+    public static class DuplicateIdMongoEntityTestTest {
+        @RegisterExtension
+        static QuarkusUnitTest runner = new QuarkusUnitTest()
+                .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                        .addClasses(DuplicateIdMongoEntity.class))
+                .withConfigurationResource("application.properties");
+
+        @Test
+        void shouldThrow() {
+            DuplicateIdMongoEntity.findAll();
+        }
+    }
+
+    @Test
+    void verifyIdTest() {
```

Comment:
see this comment https://github.com/quarkusio/quarkus/pull/8549/files#r407481562 and let me know what you think